packages/xrpc-server/src/types.ts

```diff
@@ -219,7 +219,62 @@ export type XRPCStreamHandlerConfig = {
   handler: XRPCStreamHandler
 }
 
-export { ResponseType }
+// Not all response types returned by the xrpc client are suitable to be used as HTTP status codes,
+// so we need to translate them to the server types (that are valid HTTP codes).
+const responseTypeFromClient = (status: ClientResponseType): ResponseType => {
+  switch (status) {
+    case ClientResponseType.Success:
+      return ResponseType.Success
+    case ClientResponseType.InvalidRequest:
+      return ResponseType.InvalidRequest
+    case ClientResponseType.AuthRequired:
+      return ResponseType.AuthRequired
+    case ClientResponseType.Forbidden:
+      return ResponseType.Forbidden
+    case ClientResponseType.XRPCNotSupported:
+      return ResponseType.XRPCNotSupported
+    case ClientResponseType.NotAcceptable:
+      return ResponseType.NotAcceptable
+    case ClientResponseType.PayloadTooLarge:
+      return ResponseType.PayloadTooLarge
+    case ClientResponseType.UnsupportedMediaType:
+      return ResponseType.UnsupportedMediaType
+    case ClientResponseType.RateLimitExceeded:
+      return ResponseType.RateLimitExceeded
+    case ClientResponseType.InvalidResponse:
+    case ClientResponseType.InternalServerError:
+      return ResponseType.InternalServerError
```

Comment:
If the upstream returns a 500, this server should return a 503

```suggestion
      return ResponseType.UpstreamFailure
```